### PR TITLE
Added tests for dict

### PR DESCRIPTION
**Pull request by nishnik**
Added tests for better coverage.
changed `*_eq` (* as wildcard) function (readability)

### CODE REVIEW

**Comment by abhinavagarwal07** [comment]
+1

**Comment by nishnik** [comment]
ping @isuruf

**Comment by isuruf** [comment]
Can you look into the MSVC failures?

**Comment by nishnik** [comment]
ping @isuruf

File: symengine/tests/basic/test_basic.cpp [hunk 1]
```diff
@@ -74,20 +88,151 @@ TEST_CASE("Symbol hash: Basic", "[basic]")
...
+        {
+            if (to_chk.substr(to_chk.find(key[i]) + 1, 3) != std::string(": " + val[i]))
+                return false;
+        }
+        return true;
+    };
+
+    buffer.str("");
+    buffer << ubn;
+    REQUIRE(check_map_str(buffer.str(), {"x", "y"}, {"2", "3"}));
```
Comment (isuruf) [inline symengine/tests/basic/test_basic.cpp]: You can just check for `x: 2` and `y: 3`

File: symengine/tests/basic/test_basic.cpp [hunk 2]
```diff
@@ -74,20 +88,151 @@ TEST_CASE("Symbol hash: Basic", "[basic]")
...
+    buffer.str("");
+    buffer << sb;
+    REQUIRE(buffer.str() == "[]");
+    sb.insert(i2);
+    sb.insert(y);
+
+    auto check_map_str =
+        [](std::string to_chk, std::vector<std::string> key, std::vector<std::string> val)
+    {
+        if(key.size() != val.size())
```
Comment (isuruf) [inline symengine/tests/basic/test_basic.cpp]: Space after `if`

File: symengine/tests/basic/test_basic.cpp [hunk 3]
```diff
@@ -74,20 +88,151 @@ TEST_CASE("Symbol hash: Basic", "[basic]")
...
+            return false;
+        unsigned pos;
+        for(unsigned i = 0; i < key.size(); i++)
```
Comment (isuruf) [inline symengine/tests/basic/test_basic.cpp]: Space after `for`

File: symengine/tests/basic/test_basic.cpp (hunk 1 above)
Comment (nishnik) [inline symengine/tests/basic/test_basic.cpp]: Well the algorithm kind of does the same.
It searches for occurrence of `x` and then checks that next three characters make `: 2`.

Comment (isuruf) [inline symengine/tests/basic/test_basic.cpp]: Yes, but it will fail if there is an expression like `x*y` in the map as well.

**Comment by nishnik** [comment]
ping @isuruf

**Comment by isuruf** [comment]
Coverage was improved by 0.54%. Thanks

**Pull request merged**